.github/workflows/integration-test.yml

```diff
@@ -66,6 +66,21 @@ jobs:
           setup-script-name: setup
           admin-password: admin
 
+      # https://github.com/actions/runner-images/issues/2840#issuecomment-790492173
+      - name: Remove unnecessary files Linux
+        if: ${{ runner.os == 'Linux' }}
+        run: |
+          sudo rm -rf /usr/share/dotnet
+          sudo rm -rf "$AGENT_TOOLSDIRECTORY"
+
+      - name: Remove unnecessary files Windows
+        if: ${{ runner.os == 'Windows' }}
+        run: |
+          choco uninstall julia
+          choco uninstall imagemagick.app
+          choco uninstall r.project
+          choco uninstall rtools
```

Comment:
what would be the behavior if there are more than these 4 or 1 of these doesn't exist?
Could you add a reference pointer using which you were able to list these 4?

Reply:
This is the full list I saw printed out from `choco list` on the runner

```
gradle 8.3.0
hg 5.0.0
imagemagick 7.1.1.1900
imagemagick.app 7.1.1.1900
InnoSetup 6.2.2
jq 1.7.0
julia 1.9.3
KB2919355 1.0.20160915
KB2919442 1.0.20160915
KB2999226 1.0.20181019
KB3033929 1.0.5
KB3035131 1.0.3
kubernetes-cli 1.28.2
kubernetes-helm 3.12.3
llvm 16.0.6
maven 3.8.7
mingw 12.2.0.3042023
Minikube 1.31.2
nginx 1.25.2
nodejs 18.18.0
nodejs.install 18.18.0
nssm 2.24.101.20180116
nuget.commandline 6.7.0
packer 1.9.4
php 8.2.11
pulumi 3.86.0
R.Project 4.3.1
rtools 4.3.5550
sbt 1.9.6
strawberryperl 5.32.1.1
swig 4.1.1
tortoisesvn 1.14.5.29465
vcredist140 14.36.32532
vswhere 3.1.7
wixtoolset 3.11.2
```

I haven't tested the case where one doesn't exist. I don't think it will fail the workflow, but I will check and reply back.

I didn't try to determine which of these were the largest, but I picked a handful that were pre-installed that I knew weren't required and chose to uninstall them.

Reply:
> what would be the behavior if there are more than these 4 or 1 of these doesn't exist?

It would still run if one of them didn't exist and not fail the workflow